lisa/sut_orchestrator/azure/tools.py

```diff
@@ -55,3 +57,51 @@ def get_generation(self) -> str:
         else:
             generation = "1"
         return generation
+
+
+class LisDriver(Tool):
+    """
+    This is a virtual tool to detect/install LIS drivers.
+    """
+
+    @property
+    def command(self) -> str:
+        return "modinfo hv_vmbus"
+
+    def _check_exists(self) -> bool:
+        if not self.node.os.is_posix:
+            return False
+
+        if isinstance(self.node.os, Fedora):
+            # currently LIS is only supported with Fedora
+            # and its derived distros
+            if self.node.os.package_exists(
+                "kmod-microsoft-hyper-v"
+            ) and self.node.os.package_exists("microsoft-hyper-v"):
+                return True
+        return False
+
+    def get_version(self) -> str:
+        version = ""
+        if not self._check_exists():
+            return version
+        result = self.run(' | grep -w `"version:`"')
+        if result.exit_code == 0:
+            version = result.stdout
+        return version
+
+    def install_driver(self) -> None:
+        wget_tool = self.node.tools[Wget]
+        lis_path = wget_tool.get("https://aka.ms/lis", str(self.node.working_path))
+
+        result = self.node.execute(f"tar -xvzf {lis_path}")
+        if result.exit_code != 0:
+            raise LisaException(
+                "Failed to extract tar file after downloading LIS package."
+                f"stdout: {result.stdout}"
+            )
+        lis_folder_path = str(self.node.working_path) + r"\LISISO"
```

Comment:
the path can be `self.node.working_path / "LISISO"`, but what does the "LISISO" mean?

Reply:
LISISO is the dir name that is created after extracting the tar file.

Reply:
Yes, but what the name stands for? ISO file of LIS? can it be `lis` or something easier to understand?

Reply:
The thing is tar command doesn't provide any option to rename the folder name in which it is extracting the files. I looked up for that while implementing this but didnt find anything. If you know of a way, would love to see that.

Reply:
There are plenty of posts about that (e.g. tar zxvf ugly_name.tgz --one-top-level=pretty_name), but I have not tested any :P Does this one work?

Reply:
Yup I read about this but this option is only available with tar version >= 1.29 or something. When I tested on some machines, this option wasn't available.